apps/avifenc.c

```diff
@@ -32,10 +32,12 @@ typedef struct
     avifCodecChoice codecChoice;
     int jobs;
     int targetSize;
-    avifBool qualityIsConstrained;      // true if quality explicitly set by the user
-    avifBool qualityAlphaIsConstrained; // true if qualityAlpha explicitly set by the user
+    avifBool qualityIsConstrained;        // true if quality explicitly set by the user
+    avifBool qualityAlphaIsConstrained;   // true if qualityAlpha explicitly set by the user
+    avifBool qualityGainMapIsConstrained; // true if qualityGainMap explicitly set by the user
```

Comment:
Do we want to guard that with #if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP) ?

Reply:
It's debatable, but I think having if-guards everywhere makes the code harder to read, and this variable shouldn't cause any problems if gain maps are off. You can't use the gain map api when gain maps are off anyway.

Reply:
Agreed. Plus, this feature will be official one day.